src/event/cmap.rs

```diff
@@ -117,6 +117,9 @@ pub enum ConnectionClosedReason {
     /// An error occurred while using the connection.
     Error,
 
+    /// The connection was dropped during read or write
+    Dropped,
```

Comment:
I think that adding a new reason here is the right choice, but I'm curious about whether we think it's compliant with the CMAP spec. @patrickfreed @isabelatkinson @mbroadst any thoughts on this?

Reply:
The other options don't seem to line up with what's going on, but CMAP also doesn't have any guidance for adding new reasons. We _could_ piggy back off of error here if it were important to match the spec exactly, but I have a feeling adding a new case would actually be a better API.

Reply:
a new reason seems fine to me